src/diff/json.js

```diff
@@ -11,14 +11,33 @@ jsonDiff.useLongestToken = true;
 
 jsonDiff.tokenize = lineDiff.tokenize;
 jsonDiff.castInput = function(value) {
-  return typeof value === 'string' ? value : JSON.stringify(canonicalize(value), undefined, '  ');
+  const {undefinedReplacement} = this;
+
+  return typeof value === 'string' ? value : JSON.stringify(canonicalize(value), function(k, v) {
+    if (typeof v === 'undefined') {
+      return undefinedReplacement;
+    }
+
+    return v;
+  }, '  ');
 };
 jsonDiff.equals = function(left, right) {
   return Diff.prototype.equals(left.replace(/,([\r\n])/g, '$1'), right.replace(/,([\r\n])/g, '$1'));
 };
 
-export function diffJson(oldObj, newObj, callback) { return jsonDiff.diff(oldObj, newObj, callback); }
+export function diffJson(oldObj, newObj, options, callback) {
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
 
+  if (typeof options === 'undefined') {
+    options = {};
+  }
+
+  jsonDiff.undefinedReplacement = options.undefinedReplacement;
```

Comment:
I had to assign `undefinedReplacement` directly to `jsonDiff` because `this.options` is overriden each time at the `base#diff` method (https://github.com/kpdecker/jsdiff/blob/master/src/diff/base.js#L10).